src/server/scripts/Northrend/Naxxramas/boss_heigan.cpp

```diff
@@ -176,13 +166,46 @@ class boss_heigan : public CreatureScript
 
                         eruptDirection ? ++eruptSection : --eruptSection;
 
-                        events.ScheduleEvent(EVENT_ERUPT, phase == PHASE_FIGHT ? 10000 : 3000);
+                        if (events.IsInPhase(PHASE_DANCE))
+                            events.ScheduleEvent(EVENT_ERUPT, 3 * IN_MILLISECONDS, 0, PHASE_DANCE);
+                        else
+                            events.ScheduleEvent(EVENT_ERUPT, 8 * IN_MILLISECONDS, 0, PHASE_FIGHT);
                         break;
                 }
             }
 
             DoMeleeAttackIfReady();
         }
+
+        /*void ReceiveEmote(Player* who, uint32 emoteid) override
```

Comment:
I use it locally for debugging eruption without needing to wait for dance phase. Commented out in the commit, and I'll obviously remove it when this approaches "finished".